### PR TITLE
feat: theme switching

### PR DESCRIPTION
**Pull request by rickymoorhouse**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,3 +8,4 @@ Making use of [Carbon](https://carbondesignsystem.com/) for [Hugo](https://gohug
 
  - Support for Instana EUM (just set params > instanaKey)
  - Support for theme selection (params > cdstheme) - supports `white`, `g10`, `g90` and `g100` 
+ - Support for preferred theme switching - specify lighttheme and darktheme in paramters from the options above
```
Comment (matthewfortier) [inline README.md]:
```suggestion
 - Support for preferred theme switching - specify lighttheme and darktheme in parameters from the options above
```

Comment (rickymoorhouse) [reply README.md]: Good catch - updated

File: layouts/partials/footer.html [hunk 2]
```diff
@@ -3,5 +3,26 @@
...
 </main>
 {{ partial "side-nav.html" . }}
+{{ if isset .Site.Params "darktheme" }}
+<script type="text/javascript">
+    function setTheme(themeName) {
+        document.body.className = themeName;
+    }
+    
+    function selectTheme() {
+        const prefersDarkTheme = window.matchMedia('(prefers-color-scheme: dark)');
```
Comment (matthewfortier) [inline layouts/partials/footer.html:13]: I think we could cut down on some code here with a ternary considering `prefers-color-scheme` is binary, light or dark i.e:

```js
window.matchMedia('(prefers-color-scheme: dark)').matches ? setTheme(...dark) : setTheme(...light)
```

Comment (matthewfortier) [reply layouts/partials/footer.html:13]: Not necessary obviously, works great as is :)

**Review by matthewfortier — APPROVED** [review]

**Pull request merged**